java/flight/flight-core/src/main/java/org/apache/arrow/flight/FlightService.java

```diff
@@ -51,30 +54,42 @@
 
   private final BufferAllocator allocator;
   private final FlightProducer producer;
+  private final ServerAuthHandler authHandler;
   private final ExecutorService executors;
 
-  FlightService(BufferAllocator allocator, FlightProducer producer, ExecutorService executors) {
+  FlightService(BufferAllocator allocator, FlightProducer producer, ServerAuthHandler authHandler,
+      ExecutorService executors) {
     this.allocator = allocator;
     this.producer = producer;
+    this.authHandler = authHandler;
     this.executors = new ContextPropagatingExecutorService(executors);
   }
 
-  private FlightContext makeContext(ServerCallStreamObserver<?> responseObserver) {
+  private CallContext makeContext(ServerCallStreamObserver<?> responseObserver) {
+    // Try to get the peer identity from middleware first (using the auth2 interfaces).
     final ContextAdapter contextAdapter = new ContextAdapter();
-    return new FlightContext(contextAdapter.get(AuthConstants.PEER_IDENTITY_KEY), responseObserver::isCancelled);
+    String peerIdentity = contextAdapter.get(AuthConstants.PEER_IDENTITY_KEY);
+    if (Strings.isNullOrEmpty(peerIdentity)) {
+      // Try the legacy auth interface.
+      peerIdentity = org.apache.arrow.flight.auth.AuthConstants.PEER_IDENTITY_KEY.get();
+    }
+
+    return new CallContext(peerIdentity, responseObserver::isCancelled);
   }
 
   @Override
   public StreamObserver<Flight.HandshakeRequest> handshake(StreamObserver<Flight.HandshakeResponse> responseObserver) {
-    return ServerHandshakeWrapper.wrapHandshake(responseObserver, executors);
+    // This method is not meaningful with the auth2 interfaces. Authentication would already
```

Comment:
Could this method be called by someone using Auth2 with headers? If so, how would that case be handled?

Reply:
Someone could call this with auth2 headers and if so, it'd run the auth1 workflow (which would likely be a no-op if the server was configured to use the NoOp auth handler)